### PR TITLE
Draft meeting minutes for TSC Oct19,2023

### PR DESCRIPTION
**Pull request by bigludo7**
Draft meeting minutes for TSC Oct19,2023
**Need completion & review.**

Casey could you plead update in particular for the presentation made. I did not catch the name of the gentlemen who did the presentation and probably good to add the deck presented.

### CODE REVIEW

**Comment by shilpa-padgaonkar** [comment]
@bigludo7 : Mahesh Chapalamadugu - (Verizon) End User Council representative, joined the meeting a couple of mins after the attendees were confirmed.

**Comment by hdamker** [comment]
@bigludo7 I've added my points directly into the draft, please have a look. I suppose there are still some points to add/correct.

One additional point: Could you rename the pptx TSC-2023-10-19-Release-Management.pptx so that it will be shown together with the minutes within the list?

**Review by bigludo7 — COMMENTED** [review]
Looks good for me with Herbert change.
+ changed name for the powerpoint.

**Comment by shilpa-padgaonkar** [comment]
@bigludo7 : You can kindly remove my name from Release management deck preparation line. It was all your work :)

**Review by bigludo7 — COMMENTED** [review]
Removed Shilpa from the Deck presentation team ... even if the issue raised by Shilpa was a great help ;)

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+## Release Management
+
+A deck was prepared by Ludovic/Herbert (Available within meeting notes folder)
+
+Following points were discussed:
+
+**Discussion on API versioning** – DT proposal to use alpha<n> in API path until and API (major) version is stable
+  * Distinction between repostiory release (semantic version as before) and the version shown within API base path
+    * A sub project repository can contain one or multiple (closely related) APIs
+    * The version number of releases on sub project level will be used within all APIs within the repository (= the API release belongs to vX.Y.Z of the sub project)
```
Comment (diegogonmar) [inline documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:67]: This sentence is not clear for me. I recall we said that not every API will have the same version of the release, but release will match the most advanced API in sub-project. E.g.: Release 3.2.1 with API-X in version 3.2.1, API-Y in version v2alpha1 plus in version v1.0.2, etc

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    * In addition every API within a sub project will have in the basepath
+      * the major version of the API only in case of a **stable** API (e.g. qod/v1)
+      * the major version of the API + a postfix alpha or beta, potentially with a number (e.g. qod/v1alpha3 for the third alpha version of an API)
+      * (potentially only alpha and stable will be used within CAMARA sub projects)
+    * A sub project (release) can contain
+      *  multiple APIs with different maturity, e.g. location-verification/v1 and location-retrieval/v1alpha1
+      *  multiple versions of the same API, e.g. edge-discovery/v1 (stable) and edge-discovery/v2alpha1
```
Comment (diegogonmar) [inline documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:74]: I also noted down we indicated that sub project release will remain with 0.x until first API reaches stable v1. E.g.: A release v0.1 may include an API in v1alphaX
Anyway we should further discuss on these details, once agreements are written down, small details may arise

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 1 above)
Comment (bigludo7) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: Agreed - reformulated.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 2 above)
Comment (bigludo7) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:74]: Agreed - added.

**Review by bigludo7 — COMMENTED** [review]
Fixed following @diegogonmar comments.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 2 above)
Comment (hdamker) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:74]: I suppose we need a more elaborated concept to discuss the details. The sentence was originally meant that there will be full semantic versioning only on the release of the sub project. 

But indeed, such release can contain APIs which have different major versions within the base path. To give an example:

Release 2.x.y of Device Location can contain (more complex then it most probably will be in practice, just to show the principles):
- location-verification/v1 (the first stable version, if still supported, not anymore evolved)
- location-verification/v2 (the second stable version, due to breaking changes a different API)
- location-verification/v2alpha3 (alpha version with enhancements compared to v2 which potentially will be added to the stable version within a next release of DeviceLocation)
- location-retrieval/v1 (stable version)
- location-retrieval/v2alpha1 (first alpha of next major version of location-retrieval, new major number due to lack of backward-compatibility to location-retrieval/v1)
All these API versions are belonging to the same release of the repository DeviceLocation and are unambigious defined by the commit the release is pointing to.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 1 above)
Comment (hdamker) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: I suppose we need a more elaborated concept to discuss the details. The sentence was originally meant that there will be full semantic versioning only on the release of the sub project. 

But indeed, such release can contain APIs which have different major versions within the base path. To give an example:

Release 2.x.y of Device Location can contain (more complex then it most probably will be in practice, just to show the principles):
- location-verification/v1 (the first stable version, if still supported, not anymore evolved)
- location-verification/v2 (the second stable version, due to breaking changes a different API)
- location-verification/v2alpha3 (alpha version with enhancements compared to v2 which potentially will be added to the stable version within a next release of DeviceLocation)
- location-retrieval/v1 (stable version)
- location-retrieval/v2alpha1 (first alpha of next major version of location-retrieval, new major number due to lack of backward-compatibility to location-retrieval/v1)

All these API versions are belonging to the same release of the repository DeviceLocation and are unambigious defined by the commit the release is pointing to.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 2 above)
Comment (hdamker) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:74]: yes, agreed.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md [hunk 3]
```diff
@@ -64,14 +64,15 @@ Following points were discussed:
 **Discussion on API versioning** – DT proposal to use alpha<n> in API path until and API (major) version is stable
   * Distinction between repostiory release (semantic version as before) and the version shown within API base path
     * A sub project repository can contain one or multiple (closely related) APIs
-    * The version number of releases on sub project level will be used within all APIs within the repository (= the API release belongs to vX.Y.Z of the sub project)
+    * API has not to have the same version that the family release but release will match the most advanced API in sub-project. E.g.: Release 3.0 with API-X in version 3.0, API-Y in version v2alpha1 plus in version v1.0.2, etc)
```
Comment (hdamker) [inline documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:67]: According to my comment above I propose the following changed sentence (but also propose to have further discussion on a detailed concept):

* APIs within a sub project will all belong to the sub project release, but can have have different major versions within the base path. The sub project release will match the major version of the most advanced stable API in the sub-project. E.g.: Release version will be 3.x.y if there is API-X in version v3 together with API-Y in v1 and v2alpha2.

**Review by bigludo7 — COMMENTED** [review]
Integrated @hdamker comment + @diegogonmar one on Commonalities & ICM voting.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 1 above)
Comment (bigludo7) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: @hdamker : I did not added this comment in the minutes as for me this comment should be part of the release management guidance document that you will provided - is this correct?

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+## Release Management
+
+A deck was prepared by Ludovic/Herbert (Available within meeting notes folder)
+
+Following points were discussed:
+
+**Discussion on API versioning** – DT proposal to use alpha<n> in API path until and API (major) version is stable
+  * Distinction between repostiory release (semantic version as before) and the version shown within API base path
+    * A sub project repository can contain one or multiple (closely related) APIs
+    * API has not to have the same version that the family release but release will match the most advanced API in sub-project. E.g.: Release 3.0 with API-X in version 3.0, API-Y in version v2alpha1 plus in version v1.0.2, etc)
```
Comment (shilpa-padgaonkar) [inline documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: I would like to change this line _"API has not to have the same version that the family release but release will match the most advanced API in sub-project. E.g.: Release 3.0 with API-X in version 3.0, API-Y in version v2alpha1 plus in version v1.0.2, etc)"_ 

to 

API has not to have the same version that the family release **but one of the proposal was that the release could match** the most advanced API in sub-project. E.g.: Release 3.0 with API-X in version 3.0, API-Y in version v2alpha1 plus in version v1.0.2, etc)"

As far as I remember, we decided that this needs to be thought over and then elaborated in the detailed design.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 1 above)
Comment (diegogonmar) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: API versions need to have also vx.y.x even if URL path has only major version. Anyway, I think agreed way forward in TSC is clear in minutes and we should finetune the details in the document.

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 3 above)
Comment (diegogonmar) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: Fine for me

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md (hunk 4 above)
Comment (diegogonmar) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md]: Fine for me, we indeed said we need to think a little bit more so it's ok to not set this as definitive but as first option discussed.

**Review by bigludo7 — COMMENTED** [review]
Updated following @shilpa-padgaonkar suggestion.

**Review by diegogonmar — APPROVED** [review]
LGTM

File: documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+## Release Management
+
+A deck was prepared by Ludovic/Herbert (Available within meeting notes folder)
+
+Following points were discussed:
+
+**Discussion on API versioning** – DT proposal to use alpha<n> in API path until and API (major) version is stable
+  * Distinction between repostiory release (semantic version as before) and the version shown within API base path
+    * A sub project repository can contain one or multiple (closely related) APIs
+    * API has not to have the same version that the family release , and one of the proposal was that the release could match the most advanced API in sub-project. E.g.: Release 3.0 with API-X in version 3.0, API-Y in version v2alpha1 plus in version v1.0.2, etc)"
```
Comment (eric-murray) [inline documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:67]: "API has not to have" is too strong here - it is the equivalent of the 3GPP "API shall not have", when an API could have the same version number as the family release if things work out that way.

I'd suggest to reword:
"Each API has its own version number which may be different to that of the family release. One proposal was that ..."

Comment (bigludo7) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:67]: Agreed with @eric-murray  proposal
@shilpa-padgaonkar as you provided me the sentence ok for you?
Thanks

Comment (shilpa-padgaonkar) [reply documentation/MeetingMinutes/TSC/TSC-2023-10-19-Meeting-Minutes.md:67]: @bigludo7 : This part of the sentence was not added/changed by me, I just changed the later half of it :)
But I am fine with the change proposed by Eric.

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by eric-murray — APPROVED** [review]

**Comment by hdamker** [comment]
@eric-murray 
> Can these minutes be merged? We are almost at the next meeting.
Yes, sure. I was ooo last week.

**Review by hdamker — APPROVED** [review]

**Pull request merged**